preheaders/checkout.php

```diff
@@ -552,13 +552,15 @@
 			//make the user a subscriber
 			$wpuser->set_role( get_option( 'default_role', 'subscriber' ) );
 
+			do_action( 'pmpro_checkout_before_user_auth', $user_id );

```

Comment:
```suggestion
			/**
			 * Allow hooking before the user authentication process when setting up new user.
			 *
			 * @since TBD
			 *
			 * @param int $user_id The user ID that is being setting up.
			 */
			do_action( 'pmpro_checkout_before_user_auth', $user_id );

```